### DISCUSSION TITLE
Network v16 Skyr (FVM M1) Updates and Lotus Releases Timeline Discussion

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/lotus#8346] [General Q&A] [answered]
If you have any questions & feedback wrt to the [Filecoin Network v16 Skyr (FVM M1) upgrade & Lotus releases timeline](https://github.com/filecoin-project/lotus/discussions/8347#discussion-3947806), leave a comment here!

**Comment by rjan90** [comment]
## Template Checklist / ToDo on the Butterflynetwork:

- [Tooling to perform certain tests](https://github.com/filecoin-project/lotus/issues/8483#issuecomment-1143854442).

### TSEs / Community:
- [ ] **Create BLS / secp256k1 / secp256k1-ledger:** 
- [ ] **A _send_ transaction:** 
- [ ] **Pledge a CC:** 
- [ ] **Terminate sector(s):** 
- [ ] **Compact partition(s):** 
- [ ] **Withdraw from actor:** 
- [ ] **Renew sector(s):** 
- [ ] **ChangeWorkerAddress:** 
- [ ] **ChangeOwnerAddress:** 
- [ ] **Add ControlAddresses:**
- [ ] **Create msig:**
- [ ] **Msig propose send transaction & approve:**
- [ ] **Msig propose transaction & cancel:**
- [ ] **Msig swap signers:**
- [ ] **Msig add a signer:**
- [ ] **Msig change propose threshold:**
- [ ] **Msig lock balance:**
- [ ] **Send a unverified storage deal:**
- [ ] **Assign a Notary:**
- [ ] **Notary allocate DataCap:**
- [ ] **SnapUp a sector:**
- [ ] **Send a verified storage deal:**
- [ ] **PreCommit Batch Sectors:**
- [ ] **CommitAggregate Sectors:**
- [ ] **Create a SnapShot:**
- [ ] **Run the disputer & send invalid windowPoSt:**
- [ ] **Generate and send consensus fault:**
- [ ] **Add balance to market-actor**
- [ ] **Perform a paid retrieval:**
- [ ] **Settle payment-channel:**
- [ ] **Collect payment from payment-channel:**
- [ ] **Change PeerID:**
- [ ] **Get slashed:**
- [ ] **Repay Debt:**

### Large scale testing:
_If you cannnot do the maximum, batch as many as you can and share your numbers here!_
- [ ] **Batch 256 PreCommit Sectors:**
- [ ] **CommitAggregate 819 Sectors:**
- [ ] **Batch as many PublishStorageDeals in one message:**
- [ ] **A large SubmitWindowPoSt message:**
- [ ] **A large recover message:**

### Lotus Devs/TSEs:
- [ ] **Migration succeeds for CID**
- [ ] **Get the gas execution trace and verify the gas is working as expected**
1. See the new number is reflected
2. Making sure all the ipld store are charged
- [ ] **Check CODECID things**
- [ ] **Lotus FVM-network guarded switch works**
1. Runs FVM and add debug logs in lotus 
2. Run builtin actor  and add info logs
- [ ] **Develop towards the final RC**
- [ ] **Disaster testing**

**Comment by TippyFlitsUK** [comment]
~~Noticed one change (maybe intentional??) 
A faulted sector no longer displays the remaining time available to recover, just marks it as not active:~~
```
Appears to be a result of the --fast flag. Not to worry!!

**Comment by rjan90** [comment]
## My checklist / ToDo on the Butterflynetwork:

### TSEs / Community:
- [x] **Create BLS / secp256k1 / secp256k1-ledger:** _Created BLS / secp256k1_
- [x] **A _send_ transaction:** _bafy2bzacec5527bpamzstr2ohprsvvpmqpv2653feyujxnpx7qtxqaboi7d3e_
- [x] **Pledge a CC:** _PreCommit: bafy2bzacecqhsj63w2a6qd5h3dleyr5lqbs64xnsdrnmfnf27bxdewlzw356e Commit: bafy2bzacecdid3pbzr5flkgfzka7pagf3lv2v4gvfacaxggcosmo7j7eczyek_
- [x] **Terminate sector(s):** _Sector: bafy2bzacebqrvgobzm3tc5maossskhon7wnve7dvrgoq6tx4tknvo6gjz426m Sectors:_
- [x] **Compact partition(s):** _bafy2bzaceaxm2lwoezbqi44unkf5jqa3pk4favqqpqxgyvqiujzqvq4f4cz3k_
- [x] **Withdraw from actor:** _bafy2bzacebf3t3dw3um5lflojmajo47b7zcbcdsxjp4iootaps6od3yslkjri_
- [x] **Renew sector(s):** _bafy2bzacedulhno535vagdvknbhf4yxpetmkr7ctoylhuy7onkgz5pp3r4lns_
- [x] **ChangeWorkerAddress:** _bafy2bzacebplobn6xgmzjokvpkhlmxiiapjondrziu33ka3guqxrfw6pyfjsa_
- [x] **ChangeOwnerAddress:** _First msg: bafy2bzacecrsozijv6hztuycgnup7pgdcmap5f2d7bezqioljusldhup7nliy Second msg: bafy2bzacecavgnpmqccqlyiuazjqqgifljnotxg2pfzlfg5g7k3hk5p5n4vz6_
- [x] **Add ControlAddresses:** _bafy2bzacea2uixjaees47meatxy3c2st5evoi3ggfwwlizkq7rasuodalioge_
- [x] **Create msig:** _bafy2bzaceblgulqcwdpgwhaa42zri6hg2y7dbkqq6gdwnred5ifuz5fot3ca6_
- [x] **Msig propose send transaction & approve:** _propose: bafy2bzacebuwxy6zvbzwyrkihz2tusxpgnmq2kypgaecchtjwofxa6zwq4tqg approve: bafy2bzaceb7hmtg5h6wmnzm4hgl3h4jd6s7352o3uildfu4lvmtkwvck3vx6u_
- [x] **Msig propose transaction & cancel:** _propose: bafy2bzacebodttqurh6ggspei44gfpgdr36gz23fxms7eicil6fcfkr7l65nc cancel: bafy2bzacea3xiet5t36xtox3ssahjees5oerc6lolvkj6d5tqmdtj47nwngzw_
- [x] **Msig swap signers:** _propose: bafy2bzaceczanqk5efyipgmt6ey4hjg275dnoqmpquukhony2jixlgahbuxtm approve: bafy2bzacedteacrigpuu4srrrj5agmespq7rrcey7n3ylamese4q4n4b7imhy_
- [x] **Msig add a signer:** _propose: bafy2bzacech3oikvahi3fmfm4avyqbcvbx6uqrel4orpmpcnz57d2yr2ue3wc approve: bafy2bzacedoujk5swdtb4lpb4kavqeha2gafwyamkv24pfzsnjvdf6sg4na6q_
- [x] **Msig change propose threshold:** _propose: bafy2bzaceb6cmldsouclddcox7jm7p5mqezevsck6g4pbhjqyq465do2hl344 approve: bafy2bzacedcoxr4va4bq5njbrh4vjah7lyuccghpkkgjzjmjhnfpjxg3le6ns_
- [x] **Msig lock balance:** _lock proposal: bafy2bzacebzlhqlnub55xlpvsef7jzx3x3besvc5rkodmmtxcjd4mic7svgns_
- [x] **Send a unverified storage deal:** _Proposal CID: bafyreiau3rivl4iceriiqneetz72zqudagp7jqh2t6jmhc4xj3tecmazcm_
- [x] **Assign a Notary:** _bafy2bzacec4lvjescjumllujyvvghj5jvnpvisofa5suf2hwkhgjpvwhyy3mu_
- [x] **Notary allocate DataCap:** _bafy2bzacedrcttg3dp2egfemxc5lm2u6xholcb3ihyaji6ksk6he3uet57yg4_
- [x] **Send a verified storage deal:** _bafyreigdjwdnhzf3pp5ewpiu64mpzbamjwstfg3o2557tgeq222fccol4u_
- [x] **SnapUp a sector:** _bafy2bzacebk2uqzsrsyt5hsbbee3gzsaydczbzm5ymuxjsxxo2nbclnt4egpo_
- [x] **PreCommit Batch Sectors:** _bafy2bzacebr26em52vwlsd3roqyhg4vddsem4b57j3gd4flgvq3unkw3ssqgw_
- [x] **CommitAggregate Sectors:** _bafy2bzaced3yoj5sktrkwxq43hm6eq322g2n6ctbw72vgcetf32o4b6yykddo_
- [x] **Create a SnapShot:**
- [x] **Run the disputer & send invalid windowPoSt:** _dispute msg: bafy2bzaceaqyi77nzhsedrimmggx3mskwgrlgfrjddbsex3lo2agp2htt2x6c_
- [x] **Generate and send consensus fault:** _bafy2bzacec5ssxn2cdo4v6vumqjiu3kiemi5okuz376s7txrw7r6msya6qrrs_
- [x] **Add balance to market-actor:** _bafy2bzaceaghmtkscp6xbqih67p2efmqnwhax2jmb2gaujlgt6hl27brxdnfa_
- [x] **Perform a paid retrieval:** _DealID: 1654009871143289475_
- [x] **Settle payment-channel:** _Settled t2cw6v3x2b7kyiuh56jvyawcffzz6bkaypfkzzhpa_
- [x] **Collect payment from payment-channel:** _Channel: t2cw6v3x2b7kyiuh56jvyawcffzz6bkaypfkzzhpa_
- [x] **Request multi address change:** _bafy2bzaceaa5kjyc7soeb6xrhwjhtvbecwbljbkpw5cpjecwnhkf3rxu72mxs_
- [x] **Change PeerID:** _bafy2bzacebemfmkqbyluvgmmwn4xni4san3vqath5idar6uq5f637tcukcsxo_
- [x] **Get slashed:**
- [x] **Repay Debt:** _bafy2bzaceco6b6s4j4nen5fjuok3issj64gzs6p3jhnw5hom4ibap6io3fr3q_

### Large scale testing:
- [x] **Batch 256 PreCommit Sectors:** _bafy2bzaced3zpu3d64trtokhvw6gqju6q3k4iefpmbmeajz5b3jnr3af7ue44_
- [x] **CommitAggregate 819 Sectors:** _bafy2bzaceahuepgwxxhc3kokbs27tgxuezwwobbopiivh7qg7pvtq323cflp6_
- [x] **Batch as many PublishStorageDeals in one message:** _bafy2bzacebhvdrqeoe6t22wrcby5uf5sc5q5fxcwk4visxo2x723kh2wpvsty_
- [x] **A large SubmitWindowPoSt message:** _bafy2bzaceaddbzi4sftbgs2gyw35bxmwovrmmvq2rrrt74ysmkdzevvjzkg2g_
- [x] **A large recover message:** _bafy2bzaceci72jo4v3cabwipxgg4h4hs4ts65jgrkbpgjgeuhgqwenl2zkxz4_

### Lotus Devs/TSEs:
- [ ] **Migration succeeds for CID**
- [ ] **Get the gas execution trace and verify the gas is working as expected**
1. See the new number is reflected
2. Making sure all the ipld store are charged
- [x] **Check CODECID things**
- [x] **Lotus FVM-network guarded switch works**
1. Runs FVM and add debug logs in lotus 
2. Run builtin actor  and add info logs
- [ ] **Develop towards the final RC**
- [ ] **Disaster testing**

**Reply by jennijuju** [reply]
(imma gonna hijack your checklist for the dev session!

**Comment by TippyFlitsUK** [comment]
### TSEs / Community:
- [x] **Create BLS / secp256k1 / secp256k1-ledger:** 
- [x] **A _send_ transaction:** 
- [x] **Pledge a CC:** 
- [x] **Terminate sector(s):** 
- [ ] **Compact partition(s):** 
- [x] **Withdraw from actor:** 
- [x] **Renew sector(s):** 
- [x] **ChangeWorkerAddress:** - bafy2bzacecpihjalhmefpj2qllz7s72hbputk5nym4q6cgo7eaxlwes3ed6b4
- [ ] **ChangeOwnerAddress:** 
- [ ] **Add ControlAddresses:**
- [ ] **Create msig:**
- [ ] **Msig propose send transaction & approve:**
- [ ] **Msig propose transaction & cancel:**
- [ ] **Msig swap signers:**
- [ ] **Msig add a signer:**
- [ ] **Msig change propose threshold:**
- [ ] **Msig lock balance:**
- [x] **Send a unverified storage deal:**
- [ ] **Notary allocate DataCap:**
- [ ] **Send a verified storage deal:**
- [ ] **SnapUp a sector:**
- [x] **PreCommit Batch Sectors:**
- [x] **CommitAggregate Sectors:**
- [x] **Create a SnapShot:**
- [ ] **Run the disputer & send invalid windowPoSt:**
- [ ] **Generate and send consensus fault:**
- [ ] **Perform a paid retrieval:**
- [ ] **Settle payment-channel:**
- [ ] **Collect payment from payment-channel:**
- [ ] **Change PeerID:**
- [x] **Get slashed:**
- [ ] **Repay Debt:**

### Large scale testing:
_If you cannnot do the maximum, batch as many as you can and share your numbers here!_
- [ ] **Batch 256 PreCommit Sectors:**
- [ ] **CommitAggregate 819 Sectors:**
- [ ] **Batch as many PublishStorageDeals in one message:**
- [ ] **A large SubmitWindowPoSt message:**
- [ ] **A large recover message:**

### Lotus Devs/TSEs:
- [ ] **Migration succeeds for CID**
- [ ] **Get the gas execution trace and verify the gas is working as expected**
1. See the new number is reflected
2. Making sure all the ipld store are charged
- [ ] **Check CODECID things**
- [ ] **Lotus FVM-network guarded switch works**
1. Runs FVM and add debug logs in lotus 
2. Run builtin actor  and add info logs
- [ ] **Develop towards the final RC**
- [ ] **Disaster testing**

**Comment by rjan90** [comment]
Thanks for helping out with the `Large Scale Testing` efforts https://storswift.com/ 👏💙

**Comment by namoo1** [comment]
## Meest NAM
### TSEs / Community:
- [X] Create BLS / secp256k1 / secp256k1-ledger:
- [X] A send transaction:
- [X] Pledge a CC:
- [X] Terminate sector(s):
- [ ] Compact partition(s):
- [X] Withdraw from actor:
- [ ] Renew sector(s):
- [X] ChangeWorkerAddress:
- [X] ChangeOwnerAddress:
- [X] Add ControlAddresses:
- [ ] Create msig:
- [ ] Msig propose send transaction & approve:
- [ ] Msig propose transaction & cancel:
- [ ] Msig swap signers:
- [ ] Msig add a signer:
- [ ] Msig change propose threshold:
- [ ] Msig lock balance:
- [X] Send a unverified storage deal:
- [X] Assign a Notary:
- [ ] Notary allocate DataCap:
- [X] Send a verified storage deal:
- [X] SnapUp a sector:
- [X] PreCommit Batch Sectors:
- [X] CommitAggregate Sectors:
- [X] Create a SnapShot:
- [ ] Run the disputer & send invalid windowPoSt:
- [ ] Generate and send consensus fault:
- [ ] Add balance to market-actor
- [ ] Perform a paid retrieval:
- [ ] Settle payment-channel:
- [ ] Collect payment from payment-channel:
- [X] Change PeerID:
- [ ] Get slashed:
- [ ] Repay Debt:
## Large scale testing:
If you cannnot do the maximum, batch as many as you can and share your numbers here!

- [ ] Batch 256 PreCommit Sectors:
- [ ] CommitAggregate 819 Sectors:
- [ ] Batch as many PublishStorageDeals in one message:
- [ ] A large SubmitWindowPoSt message:
- [ ] A large recover message:
## Lotus Devs/TSEs:
- [ ] Migration succeeds for CID
- [ ] Get the gas execution trace and verify the gas is working as expected
1. See the new number is reflected
2. Making sure all the ipld store are charged
- [ ] Check CODECID things
- [ ] Lotus FVM-network guarded switch works
1. Runs FVM and add debug logs in lotus
2. Run builtin actor and add info logs
- [ ] Develop towards the final RC
- [ ] Disaster testing

**Comment by TippyFlitsUK** [comment]
Checklist for NV19 & NV20

```
Daemon:  1.23.0+mainnet+git.d1d4b35ad+api1.5.0
Local: lotus-miner version 1.23.0+mainnet+git.d1d4b35ad
```

### TSEs / Community:
- [x] **Create BLS / secp256k1 / secp256k1-ledger:** - bafy2bzacedcfzvitrchkbqrewkt5lts6clh5by5jfzt4i56jvv2zbjnnxkqku
- [x] **A _send_ transaction:** - bafy2bzacecyzufqnhuyznmzvis67iogcxl26gi4apfi6deyifne4hjv6rb3qu
- [x] **Pledge a CC:** 
- [x] **Terminate sector(s):** 
- [ ] **Compact partition(s):** 
- [x] **Withdraw from actor:** - bafy2bzacebuj7icowymdtds45bkv2phe664ffl2vekzwrq73jqxmhnr7tssww
- [x] **Renew sector(s):** - bafy2bzacedds3ccwpxud5g6zrxhtfbvonujnsunaipdiwvnp6n3wnf6zhqyam
- [x] **ChangeWorkerAddress:** - bafy2bzacecpihjalhmefpj2qllz7s72hbputk5nym4q6cgo7eaxlwes3ed6b4
- [x] **ChangeOwnerAddress:**  - bafy2bzaceaiynwytpfwtlv76bynlyifj6cd7lhzyjvfpcsvqdneq6cgczerbq
- [x] **Add ControlAddresses:** - bafy2bzacebvjrwiklz4hzcykc35vhmjscdjf6r7ttjypnjgewi3awnvgz6vn6
- [x] **Create msig:** - bafy2bzacedcfzvitrchkbqrewkt5lts6clh5by5jfzt4i56jvv2zbjnnxkqku
- [ ] **Msig propose send transaction & approve:**
- [ ] **Msig propose transaction & cancel:**
- [ ] **Msig swap signers:**
- [ ] **Msig add a signer:**
- [ ] **Msig change propose threshold:**
- [ ] **Msig lock balance:**
- [x] **Send a unverified storage deal:**
- [ ] **Assign a Notary:**
- [ ] **Notary allocate DataCap:**
- [ ] **SnapUp a sector:**
- [ ] **Send a verified storage deal:**
- [x] **PreCommit Batch Sectors:**
- [x] **CommitAggregate Sectors:**
- [x] **Create a SnapShot:**
- [ ] **Run the disputer & send invalid windowPoSt:**
- [ ] **Generate and send consensus fault:**
- [ ] **Add balance to market-actor**
- [ ] **Perform a paid retrieval:**
- [ ] **Settle payment-channel:**
- [ ] **Collect payment from payment-channel:**
- [ ] **Change PeerID:**
- [ ] **Get slashed:**
- [ ] **Repay Debt:**

### Large scale testing:
_If you cannnot do the maximum, batch as many as you can and share your numbers here!_
- [ ] **Batch 256 PreCommit Sectors:**
- [ ] **CommitAggregate 819 Sectors:**
- [ ] **Batch as many PublishStorageDeals in one message:**
- [ ] **A large SubmitWindowPoSt message:**
- [ ] **A large recover message:**

### Lotus Devs/TSEs:
- [ ] **Migration succeeds for CID**
- [ ] **Get the gas execution trace and verify the gas is working as expected**
1. See the new number is reflected
2. Making sure all the ipld store are charged
- [ ] **Check CODECID things**
- [ ] **Lotus FVM-network guarded switch works**
1. Runs FVM and add debug logs in lotus 
2. Run builtin actor  and add info logs
- [ ] **Develop towards the final RC**
- [ ] **Disaster testing**

**Reply by rjan90** [reply]
## My checklist / ToDo on a localnet:

### TSEs / Community:
- [x] **Create BLS / secp256k1 / secp256k1-ledger/delegated:**
- [x] **A _send_ transaction:** bafy2bzacebvnunv6rieb4s7hibxks2i54d3vh2k4mejk43erjamls64qgnb2k
- [x] **Pledge a CC:** bagboea4b5abcah4gkuk6k6r4yygq42swgmerqoillhgjbgc65q2dlsmbhzwf4q3l
- [x] **Terminate sector(s):** 
- [ ] **Compact partition(s):** 
- [x] **Withdraw from actor:** 
- [x] **Renew sector(s):** 
- [x] **ChangeWorkerAddress:** 
- [x] **ChangeOwnerAddress:** 
- [x] **Add ControlAddresses:** 
- [x] **Propose and change beneficiary**
- [x] **Create msig:** 
- [x] **Msig propose send transaction & approve:** 
- [x] **Msig propose transaction & cancel:** 
- [x] **Msig swap signers:** 
- [x] **Msig add a signer:** 
- [x] **Msig change propose threshold:** 
- [x] **Msig lock balance:** 
- [x] **Send a unverified storage deal:** 
- [x] **Assign a Notary:** 
- [x] **Notary allocate DataCap:** 
- [x] **Send a verified storage deal:** 
- [x] **SnapUp a sector:** 
- [x] **PreCommit Batch Sectors:** 
- [x] **CommitAggregate Sectors:** 
- [x] **Create a SnapShot:**
- [x] **Run the disputer & send invalid windowPoSt:** Disputer running:
2023-04-25T07:56:50.111-0400    INFO    disputer        cli/disputer.go:227     last checked epoch      {"epoch": "1650"}
   - Its not disputing any regular windowPoSt.
- [ ] **Generate and send consensus fault:** 
- [x] **Let a single sector fail in a windowPoSt**
   - [x] **Successfully recover that sector in next windowPoSt**
- [x] **Add balance to market-actor:** 
- [x] **Withdraw funds from the market-actor:**
- [ ] **Perform a paid retrieval:** 
- [x] **Settle payment-channel:** 
- [x] **Collect payment from payment-channel:** 
- [ ] **Request multi address change:** 
- [ ] **Change PeerID:** 
- [ ] **Get slashed:**
- [ ] **Repay Debt:** 

### Large scale testing:
- [ ] **Batch 256 PreCommit Sectors:** 
- [ ] **CommitAggregate 819 Sectors:** 
- [ ] **Batch as many PublishStorageDeals in one message:** 
- [ ] **A large SubmitWindowPoSt message:** 
- [ ] **A large recover message:**

**Comment by TippyFlitsUK** [comment, marked as answer]
NV16 has successfully passed.
